R/arc-select.R

```diff
@@ -224,26 +233,27 @@ collect_layer <- function(
   # count the number of features in a query
   n_feats <- count_results(req, query_params)
 
-  if (is.null(n_feats)) {
-    cli::cli_abort(c(
-      "Cannot determine the number of features in request.",
-      "i" = "did you set custom parameters via {.arg ...}?"
-    ), call = error_call)
+  if (is.null(n_feats) && is.null(n_max)) {
```

Comment:
The `count_result()` function is used to determine how many pages to create and iterate through. When the `count_results()` fails, that implies that the subsequent queries will also fail. It will return a NULL then the check for `n_feats > n_max` will evaluate to `logical(0)` which causes an error. So i think this check _needs_ to be for just `is.null(n_feats)`

Reply:
Current PR behavior 
```r
furl <- "https://services3.arcgis.com/ZvidGQkLaDJxRSJ2/arcgis/rest/services/PLACES_LocalData_for_BetterHealth/FeatureServer/0"

arc_open(furl) |> 
  arc_select(spatialRel = "wouldnt you like to know?", n_max = 10)
```

<img width="631" alt="image" src="https://github.com/R-ArcGIS/arcgislayers/assets/13534508/b9d1dd19-0658-4ad3-974a-cdb5563e299e">

After removing the second check: 
```suggestion
  if (is.null(n_feats)) {
```

<img width="721" alt="image" src="https://github.com/R-ArcGIS/arcgislayers/assets/13534508/2a3ed196-9d62-4b1c-8e68-e4befa041ab9">